### ISSUE TITLE
exception on two transitions to the same deep history node

### ISSUE THREAD

**Issue by fjouault** [plantuml/plantuml#1788]
**Describe the bug**
The following PlantUML code yields an exception with versions >= v1.2023.2, but works as expected with versions <= v1.2023.1:

```plantuml
@startuml
state B
A --> B[H*] : ev1
A --> B[H*] : ev2
@enduml
```

There is no issue with regular history nodes, only with deep history nodes.

For instance, with version v1.2023.2, the exception is:
```cy
Error java.lang.IllegalStateException
java.lang.IllegalStateException
	at net.atmp.CucaDiagram.reallyCreateLeaf(CucaDiagram.java:180)
	at net.sourceforge.plantuml.statediagram.StateDiagram.getDeepHistory(StateDiagram.java:164)
	at net.sourceforge.plantuml.statediagram.command.CommandLinkStateCommon.getEntity(CommandLinkStateCommon.java:165)
	at net.sourceforge.plantuml.statediagram.command.CommandLinkStateCommon.getEntityEnd(CommandLinkStateCommon.java:151)
	at net.sourceforge.plantuml.statediagram.command.CommandLinkStateCommon.executeArg(CommandLinkStateCommon.java:82)
	at net.sourceforge.plantuml.statediagram.command.CommandLinkStateCommon.executeArg(CommandLinkStateCommon.java:60)
	at net.sourceforge.plantuml.command.SingleLineCommand2.execute(SingleLineCommand2.java:150)
	at net.sourceforge.plantuml.command.ProtectedCommand.execute(ProtectedCommand.java:57)
	at net.sourceforge.plantuml.AbstractPSystem.executeCommand(AbstractPSystem.java:166)
	at net.sourceforge.plantuml.command.PSystemCommandFactory.executeFewLines(PSystemCommandFactory.java:125)
	at net.sourceforge.plantuml.command.PSystemCommandFactory.createSystem(PSystemCommandFactory.java:108)
	at net.sourceforge.plantuml.PSystemBuilder.createPSystem(PSystemBuilder.java:145)
	at net.sourceforge.plantuml.BlockUml.getDiagram(BlockUml.java:187)
	at net.sourceforge.plantuml.SourceFileReaderAbstract.getGeneratedImages(SourceFileReaderAbstract.java:170)
	at net.sourceforge.plantuml.Run.manageFileInternal(Run.java:524)
	at net.sourceforge.plantuml.Run.processArgs(Run.java:407)
	at net.sourceforge.plantuml.Run.manageAllFiles(Run.java:374)
	at net.sourceforge.plantuml.Run.main(Run.java:208)
You should send a mail to plantuml@gmail.com or post to https://plantuml.com/qa with this log (V1.2023.2)
Error line 4 in file: build/conceptionGenerale-gui-SM.puml
Some diagram description contains errors
```

**To Reproduce**
Try to render the above code.

**Expected behavior**
No exception and correct rendering, as with older PlantUML versions.

**Screenshots**
Expected:
![image](https://github.com/plantuml/plantuml/assets/10452457/f40c42d8-f4a5-4eb8-b2b1-4abf9faab50d)

Actual (with version v1.2024.5beta3):
![image](https://github.com/plantuml/plantuml/assets/10452457/3f9d90a3-e064-4ac7-a2b8-e7f7f7dcc912)

**Desktop (please complete the following information):**

- OS: a priori irrelevant, tested under Linux & in the Web interface.
- Browser: a priori irrelevant, tested with CLI & Chrome
- Versions >= v1.2023.2

**Additional context**
The `StateDiagram::getDeepHistory` methods were changed between versions v1.2023.1 and v1.2023.2 to use `reallyCreateLeaf` instead of `getOrCreateLeaf`.

Because regular history nodes do not have the issue, comparing the code of `getHistorical(String idShort)` and `getDeepHistory(String idShort)` could help understand the issue.
The only significant differences that I can see are:
* the second argument to `reallyCreateLeaf` is `Display.getWithNewlines(ident)` for `getHistorical` and `Display.getWithNewlines("")` for `getDeepHistory`,
* `cleanId` is used in `getDeepHistory` but not in `getHistorical`.

**Comment by The-Lum** [comment]
Hello all,

_Awaiting correction..._

_[Just for the record]_
See also this similar corrected issue or request _(but for regular history)_:
- https://forum.plantuml.net/18282/bug-only-one-transition-permitted-to-state-in-state-diagram

Now we observe the issue for deep history...

Regards,
Th.

**Comment by The-Lum** [comment]
Hi all,

_[This is an Issue Review]_ 👀 
This is now fixed, on [v1.2024.8](https://github.com/plantuml/plantuml/releases/tag/v1.2024.8).
- Thanks @fjouault for the report.
- Thanks PlantUML team for your work.

Regards,
Th.
